Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Favorites/AssetBrowserFavoritesManager.cpp

```diff
@@ -0,0 +1,323 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "AssetBrowserFavoritesManager.h"
+
+#include <AzCore/Utils/Utils.h>
+#include <AzCore/Module/Environment.h>
+#include <AzCore/std/smart_ptr/make_shared.h>
+
+#include <AzFramework/Asset/AssetCatalogBus.h>
+
+#include <AzToolsFramework/AssetBrowser/AssetBrowserBus.h>
+#include <AzToolsFramework/AssetBrowser/AssetBrowserEntry.h>
+#include <AzToolsFramework/AssetBrowser/AssetBrowserModel.h>
+#include <AzToolsFramework/AssetBrowser/Entries/AssetBrowserEntryCache.h>
+#include <AzToolsFramework/AssetBrowser/Entries/AssetBrowserEntryUtils.h>
+#include <AzToolsFramework/AssetBrowser/Entries/FolderAssetBrowserEntry.h>
+#include <AzToolsFramework/AssetBrowser/Favorites/AssetBrowserFavoriteItem.h>
+#include <AzToolsFramework/AssetBrowser/Favorites/EntryAssetBrowserFavoriteItem.h>
+#include <AzToolsFramework/AssetBrowser/Favorites/SearchAssetBrowserFavoriteItem.h>
+#include <AzToolsFramework/AssetBrowser/Favorites/AssetBrowserFavoritesModel.h>
+#include <AzToolsFramework/AssetBrowser/Favorites/AssetBrowserFavoritesView.h>
+#include <AzToolsFramework/AssetBrowser/Search/SearchWidget.h>
+#include <AzToolsFramework/AssetBrowser/Views/AssetBrowserViewUtils.h>
+
+#include <QModelIndex>
+#include <QSettings>
+
+namespace AzToolsFramework
+{
+    namespace AssetBrowser
+    {
+        const char* AssetBrowserFavoritesManager::s_environmentVariableName = "AssetBrowserFavoritesManager";
+        AZ::EnvironmentVariable<AssetBrowserFavoritesManager*> AssetBrowserFavoritesManager::g_globalInstance;
+
+        AssetBrowserFavoritesManager::AssetBrowserFavoritesManager()
+        {
+            AssetBrowserFavoriteRequestBus::Handler::BusConnect();
+            AssetBrowserComponentNotificationBus::Handler::BusConnect();
+            AzFramework::AssetCatalogEventBus::Handler::BusConnect();
+        }
+
+        AssetBrowserFavoritesManager ::~AssetBrowserFavoritesManager()
+        {
+            AzFramework::AssetCatalogEventBus::Handler::BusDisconnect();
+            AssetBrowserComponentNotificationBus::Handler::BusDisconnect();
+            AssetBrowserFavoriteRequestBus::Handler::BusDisconnect();
+        }
+
+        void AssetBrowserFavoritesManager::AddFavoriteItem(AssetBrowserFavoriteItem* item)
+        {
+            m_favorites.push_back(item);
+
+            SaveFavorites();
+        }
+
+        void AssetBrowserFavoritesManager::AddFavoriteAsset(const AssetBrowserEntry* favorite)
```

Comment:
We should either prevent product assets from being favorited, or add the source asset it belongs to instead. This is in line with the rest of the context menu actions, since you cannot perform file operations on product assets.

Reply:
You can add a product as a favorite, but you can only perform the same operations that you would be able to in the rest of the asset browser.